docs/release/controllers.md

```diff
@@ -0,0 +1,178 @@
+Stable API versions are not subject to removal in any future release of a controller major version.
+
+In effect, this means that for as long as Flux `v2` is being maintained, all the stable API versions 
+will be supported.
+
+## Controller versioning
+
+The Flux controllers and their Go API packages are released by following the
+[Go module version numbering](https://go.dev/doc/modules/version-numbers) conventions:
+
+- `vX.Y.Z-RC.W` release candidates e.g. `v1.0.0-RC.1`
+- `vX.Y.Z` stable releases e.g. `v1.0.0`
+
+The API versioning and controller versioning are indirectly related. For example,
+a source-controller minor release `v1.1.0` can introduce a new API version
+`v1beta1` for a Kind `XRepository` in the `source.toolkit.fluxcd.io` group.
+
+### Release candidates
+
+Release candidates are intended for testing new features or improvements before a final release.
+
+In most cases, a maintainer will publish a release candidate of a controller for Flux users
+to tests it on their staging clusters. Release candidates are not meant to be deployed in production
+unless advised to do so by a maintainer.
+
+### Patch releases
+
+Patch releases are intended for critical bug fixes to the latest minor version, such as addressing security
+vulnerabilities or fixes to severe problems with no workaround.
+
+Patch releases do not contain breaking changes, feature additions or any type of user-facing changes.
+If a CVE fix requires a breaking change, then a minor release will provide the fix.
+
+We expect users to be running the latest patch release of a given minor release as soon as the
+controller release is included in a Flux patch release.
+
+### Minor releases
+
+Minor releases are intended for backwards compatible feature additions and improvements.
+Note that breaking changes may occur if required by a security vulnerability fix.
+
+Minor releases are used when updating Kubernetes dependencies such as `k8s.io/api` from one minor version to another.
+
+In effect, this means a minor version will be released for all Flux controllers approximately every four months
+after each Kubernetes minor version release. To properly validate the controllers against the latest Kubernetes version,
+we reserve a time window of at least two weeks for Flux controllers end-to-end testing. 
+
+### Major releases
+
+Major releases are intended for drastic changes in the controller behaviour or security stance.
+
+A controller major release will be announced ahead of time throughout all communication channels,
+and a support window of one year will be provided for the previous major version.
+
+## Release cadence
+
+Flux controllers follow Kubernetes three releases per year cadence. After each Kubernetes minor release,
+all controllers are tested against the latest Kubernetes version and are released at approximately two
+weeks after Kubernetes. The newly released controllers offer support for Kubernetes N-2 minor versions.
```

Comment:
We probably highlight that we follow Kubernetes' support policy, which at the moment is N-2 minor versions.